src/js/media-container.js

```diff
@@ -145,6 +145,94 @@ template.innerHTML = `
 
 const MEDIA_UI_ATTRIBUTE_NAMES = Object.values(MediaUIAttributes);
 
+// Watch for child adds/removes and update the media element if necessary
+const mutationCallback = (mutationsList) => {
+  for (let mutation of mutationsList) {
+    /** @type MediaContainer */
+    const container = mutation.target;
+    const media = container.media;
+
+    if (mutation.type === 'childList') {
+      // Media element being removed
+      mutation.removedNodes.forEach((node) => {
+        // Is this a direct child media element of media-controller?
+        // TODO: This accuracy doesn't matter after moving away from media attrs.
+        // Could refactor so we can always just call 'dispose' on any removed media el.
+        if (node.slot == 'media') {
+          // Check if this was the current media by if it was the first
+          // el with slot=media in the child list. There could be multiple.
+          let previousSibling =
+            mutation.previousSibling &&
+            mutation.previousSibling.previousElementSibling;
+
+          // Must have been first if no prev sibling or new media
+          if (!previousSibling || !media) {
+            container.mediaUnsetCallback(node);
+          } else {
+            // Check if any prev siblings had a slot=media
+            // Should remain true otherwise
+            let wasFirst = previousSibling.slot !== 'media';
+            while (
+              (previousSibling = previousSibling.previousSibling) !== null
+            ) {
+              if (previousSibling.slot == 'media') wasFirst = false;
+            }
+            if (wasFirst) container.mediaUnsetCallback(node);
+          }
+        }
+      });
+
+      // Controls or media element being added
+      // No need to inject anything if media=null
+      if (media) {
+        mutation.addedNodes.forEach((node) => {
+          if (node == media) {
+            // Update all controls with new media if this is the new media
+            container.handleMediaUpdated(media).then((media) =>
+              container.mediaSetCallback(media)
+            );
+          }
+        });
+      }
+    }
+  }
+};
+
+const defaultBreakpoints = 's:0 sm:384 m:576 ml:960 l:1280 xl:1600 fhd:1920 qhd:2560 uhd:3840';
```

Comment:
Proposal: `media-container-breakpoints` as the attribute to configure this

Reply:
`container-breakpoints` or `containerbreakpoints` ?

Reply:
we really need to come up with a style guide for media-chrome that lays out the options explicitly, so, we don't need to think about it each time.